### ISSUE TITLE
Feature Request: Fullscreen mode

### ISSUE THREAD

**Issue by niklasgrewe** [ONLYOFFICE/onlyoffice-nextcloud#395]
Hi, thanks for this great integration into nextcloud. I love it. But it is possible to use it in fullscreen mode, without the blue nextcloud banner at top of the page? 

![image](https://user-images.githubusercontent.com/61348836/98081074-c0875180-1e76-11eb-8e99-2b4a980a4fe2.png)
> onlyoffice in nextcloud with banner

![image](https://user-images.githubusercontent.com/61348836/98081196-f7f5fe00-1e76-11eb-9d39-4f7e5ac81426.png)
> onlyoffice in seafile without any banner

I know that i can open documents in new tab, but the blue nextcloud banner is always visible. I would like to see an option in settings, so i can easily switch on/off the fullscreen mode, to see more of my document.

**Comment by axheli** [comment]
Yes, I agree with this. It would also be nice to have a X icon to close the document.

**Comment by violoncelloCH** [comment]
related to #445

**Comment by axheli** [comment]
would this be related #24889 ?

**Comment by urnash** [comment]
> A quick & dirty hack to make this work is to use the JSLoader App (from the app store) to inject this code in the page:
> 
> ```
> if ($('.app-onlyoffice').length || $('#onlyofficeFrame').lenght) {
>     $('#header').remove();
>     $('#content').attr('id', 'alt_content');
> }
> ```
> 
> After you open the document, refresh the page once more and it will go fullscreen. Not the prettiest way but it can work until we get a real implementation

Unfortunately, this does not work correctly, wrong top and bottom padding.
But even if they were correct, the page size is shorter than necessary.

![Screenshot_16](https://user-images.githubusercontent.com/26555345/205450014-88cd1bb3-eec9-41e3-bcc1-43be76ed3d43.png)

**Comment by wrongt** [comment]
appending this to quick & dirty hack will fix the top
`$('#alt_content').css({"min-height": "calc(100% - 1px)", "margin-top": "50px"});`
and this 
`$('iframe[name="frameEditor"]').css({"height":"100%"});`
is supposed to fix the bottom, but didn't work ...

**Comment by tzxmaster** [comment]
I've written this small snippet if you want to test:

```
const button = document.createElement('button');
button.innerText = 'Fullscreen';
document.getElementById("header").appendChild(button);

button.addEventListener('click', () => {
  var elem = document.getElementById("onlyofficeFrame"); 
	elem.requestFullscreen();
})
```
It will add a new button in upper right corner and if you click on it it will make the page fullscreen.

You can test it using the console.

note: if you refresh the page (F5) while a document is open it will not work anymore because the "onlyofficeFrame" element will not be present anymore.

I've tested it with JSLoader app as well and it works.

**Comment by tzxmaster** [comment]
I've updated the code to work after refresh as well:

```
const button = document.createElement('button')
button.innerText = 'Fullscreen'
document.getElementById("header").children[1].prepend(button)

button.addEventListener('click', () => {
    var elem = document.getElementById("onlyofficeFrame");
    if (elem == null) {
        elem = document.getElementById("app-content");
    }
    if (elem == null) {
        elem = document.getElementById("app");
    }
    elem.requestFullscreen();
})
```
It will work with any app as well, not just for ONLYOFFICE.
I've also moved the button to a better place.

I am very bad at UI/frontend stuff so if some CSS/JS guru can help us with a nice icon instead of a button, it would be nice.

**Comment by vagner-dias** [comment]
+1

Perhaps this could be implemented in the OnlyOffice NC App ? >> https://github.com/ONLYOFFICE/onlyoffice-nextcloud/issues/764

**Comment by vagner-dias** [comment]
Hello, Folks.
I found this commit from collabora using the keyword allowfullscreen , and I think this could help fix the issue. I just don´t know exactly how this could be implemented.

https://github.com/nextcloud/richdocuments/commit/ee5491b41a1718c94d1770929e49e49dbdaf565a#diff-982c3683c43a8fb4c83489c33d580c6d51f9a304c76da6cf5380f351b18f9180

Would this help somehow?

**Comment by wrongt** [comment]
Script for JSLoader App that remove nextcloud header in shared application (for collaboration via public links)
```
//Remove header in fileshare application
if ($('#files-public-content').length) {
    $('#onlyofficeFrame').on('load', function () {
        $('#header').remove();
        $('#content').attr('id', 'alt_content');
        $('#alt_content').css({ "min-height": "calc(100% - 1px)", "margin-top": "50px" });

        var ifr = document.querySelectorAll('#onlyofficeFrame');
        ifr[0].style.cssText += 'height: calc(100vh - -57px);';
    });
}
```

**Comment by guojiaqiang89** [comment]
> A quick & dirty hack to make this work is to use the JSLoader App (from the app store) to inject this code in the page:
> 
> ```
> if ($('.app-onlyoffice').length || $('#onlyofficeFrame').lenght) {
>     $('#header').remove();
>     $('#content').attr('id', 'alt_content');
> }
> ```
> 
> After you open the document, refresh the page once more and it will go fullscreen. Not the prettiest way but it can work until we get a real implementation

It work good on Nextcloud23, When I change to Nextcloud 27, It dose not work.

**Comment by emes81** [comment]
I'd like to also add support for this; the upper-left link to the OnlyOffice website should be replaced with a link to return to the Nextcloud host

**Comment by Plinsboorg** [comment]
+1 for this feature

**Comment by SHU-red** [comment]
Also would like to see this

**Comment by emes81** [comment]
Another vote for this. For a more substantial reason, also. 

Nextcloud currently recommends setting the "X-Frame-Options" HTTP header to "SAMEORIGIN". If not, the admin settings display this warning:

`The "X-Frame-Options" HTTP header is not set to "SAMEORIGIN". This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly.`

However, adjusting that setting means that OnlyOffice can no longer be embedded in NC. More here:

https://forum.onlyoffice.com/t/nextcloud-28-0-2-onlyoffice-nginx-onlyoffice-inacessible-when-x-frame-options-set-to-sameorigin-always/7785/5

Collabora does not have this issue in Nextcloud. Hence, an option to remove the Nextcloud banner would not only be better from a UI perspective: it might also allow OnlyOffice to match current Nextcloud security standards.

**Comment by AlphaCraft9658** [comment]
I support this feature. There should be a fullscreen button or even a "Hide Nextcloud UI" button. The fullscreen button would enter fullscreen, hiding the Nextcloud UI and the browser UI, and the "Hide Nextcloud UI" button would just hide the top bar of Nextcloud and remove the border radius of the OnlyOffice wrapper/iframe so that all the UI elements of OnlyOffice are unobstructed.

**Comment by chief-nerd** [comment]
would support this

**Comment by AlphaCraft9658** [comment]
Updates?

**Comment by roschi02** [comment]
Sadly, the scripts here didn’t work for me, so I wrote my own with some help from AI. I just wanted to share it in case someone else runs into the same problem.

```
(function() {
  const state = {
    applied: false,
    originalStyles: new Map()
  };

  function saveOriginalStyle(el) {
    if (!el || state.originalStyles.has(el)) return;
    state.originalStyles.set(el, el.getAttribute('style'));
  }

  function restoreOriginalStyles() {
    state.originalStyles.forEach((style, el) => {
      if (!el) return;
      if (style === null) el.removeAttribute('style');
      else el.setAttribute('style', style);
    });
    state.originalStyles.clear();
    state.applied = false;
  }

  function applyOnlyOfficeFullScreen() {
    const iframeElement = document.querySelector('iframe[src*="onlyoffice"]');
    if (!iframeElement) {
      if (state.applied) restoreOriginalStyles();
      return;
    }

    const headerElement = document.getElementById('header');
    const contentVueElement = document.getElementById('content-vue');

    if (!state.applied) {
      saveOriginalStyle(iframeElement);
      saveOriginalStyle(headerElement);
      saveOriginalStyle(contentVueElement);
    }

    // Set OnlyOffice iframe to full screen
    iframeElement.style.top = '0';
    iframeElement.style.height = '100vh';
    iframeElement.style.margin = '0';
    iframeElement.style.zIndex = '2000';

    // Hide the header
    if (headerElement) headerElement.style.display = 'none';

    // Adjust the content container
    if (contentVueElement) {
      contentVueElement.style.top = '0';
      contentVueElement.style.width = '100vw';
      contentVueElement.style.height = '100vh';
      contentVueElement.style.margin = '0';
      contentVueElement.style.borderRadius = '0';
    }

    state.applied = true;
  }

  const targetNode = document.querySelector('.app-content') || document.body;
  const observer = new MutationObserver(applyOnlyOfficeFullScreen);
  observer.observe(targetNode, { childList: true, subtree: true });

  // Execute once on page load
  window.addEventListener('load', applyOnlyOfficeFullScreen);
})();
```

**Comment by riparise** [comment]
Instead o injecting a js, one can use the "custom css" app, and the following:

```ccc
body:has(.onlyoffice-iframe-container) #header {
    display: none !important;
}

body:has(.onlyoffice-iframe-container) #content-vue {
    margin-top: var(--body-container-margin) !important;
    height: calc(100% - var(--body-container-margin) * 2) !important;
}

body:has(.onlyoffice-iframe-container) #app-public {
    height: calc(100dvh - var(--body-container-radius)) !important;
}

body:has(.onlyoffice-iframe-container) #app{
    padding-top: 0 !important;
}

body:has(iframe[name="frameEditor"]) iframe[name="frameEditor"] {
    height: calc(100dvh - var(--body-container-radius)) !important;
}
```

again, hacky and not robust, but gets the job done (for now)
